core/src/main/scala/org/apache/spark/streaming/SnappyStreamingContext.scala

```diff
@@ -307,7 +341,23 @@ object SnappyStreamingContext extends Logging {
       ): SnappyStreamingContext = {
     val checkpointOption = CheckpointReader.read(
       checkpointPath, new SparkConf(), hadoopConf, createOnError)
-    checkpointOption.map(new SnappyStreamingContext(null, _, null)).getOrElse(creatingFunc())
+    checkpointOption.map(new SnappyStreamingContext(null, _, null)).
+      getOrElse(creatingFunc())
+
+  }
+
+  def getOrCreateWithUseCredential(
```

Comment:
What is this function for?

Reply:
I have added a doc comment. This function is to fix Snap-2153 where a streaming context created from check point does not have user credentials passed to the job.